packages/admin/admin/src/form/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,237 @@
+            gap: "10px",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+            border: `1px dashed ${palette.error.main}`,
+            color: palette.error.main,
+        },
+        errorMessage: {
+            display: "flex",
+            alignItems: "center",
+            alignSelf: "stretch",
+            color: palette.error.main,
+            gap: "5px",
+        },
+        disabled: {},
+        error: {},
+    });
+};
+
+export interface FinalFormFileUploadProps extends FieldRenderProps<File | File[], HTMLInputElement> {
+    disableDropzone?: boolean;
+    disableButton?: boolean;
+    accept: Accept;
+    maxSize: number;
+}
+
+const FinalFormFileUploadComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormFileUploadProps> = ({
+    classes,
+    disabled,
+    disableDropzone,
+    disableButton,
+    accept,
+    maxSize = 50 * 1024 * 1024,
+    input: { onChange, value: fieldValue, multiple: multipleFiles },
+}) => {
+    const onDrop = React.useCallback(
+        (acceptedFiles: File[]) => {
+            multipleFiles && Array.isArray(fieldValue) ? onChange([...fieldValue, ...acceptedFiles]) : onChange([...acceptedFiles]);
+        },
+        [fieldValue, multipleFiles, onChange],
+    );
+
+    const removeFile = (removedFile: File) => () => {
+        const newFiles = Array.isArray(fieldValue) ? fieldValue.filter((file) => file !== removedFile) : undefined;
+        onChange(newFiles);
+    };
+
+    const { fileRejections, getRootProps, getInputProps, isDragReject } = useDropzone({
+        onDrop,
+        accept,
+        disabled,
+        multiple: multipleFiles,
+        maxSize: maxSize,
+    });
+
+    // list of the accepted files
+    const files = Array.isArray(fieldValue)
+        ? fieldValue.map((file) => (
+              <div key={file.name} className={classes.fileListItem}>
+                  {file.name.length < 20 ? file.name : `${file.name.substring(0, 20)}...`}
+                  <div>
+                      <Chip label={<PrettyBytes value={file.size} />} />
+                      <IconButton onClick={removeFile(file)}>
+                          <Delete />
+                      </IconButton>
+                  </div>
+              </div>
+          ))
+        : !Array.isArray(fieldValue) &&
+          fieldValue.name !== undefined && (
+              <div key={fieldValue.name} className={classes.fileListItem}>
+                  {fieldValue.name.length < 20 ? fieldValue.name : `${fieldValue.name.substring(0, 20)}...`}
+                  <div>
+                      <Chip label={<PrettyBytes value={fieldValue.size} />} />
+                      <IconButton onClick={removeFile(fieldValue)}>
+                          <Delete />
+                      </IconButton>
+                  </div>
+              </div>
+          );
```

Comment:
Please find a way to not duplicate the rendering logic. Something like

```tsx
let files;

if(Array.isArray(fieldValue)) {
  files = fieldValue;
} else if(fieldValue.name !== undefined) {
  files = [fieldValue];
} else {
  files = []
}
```

You can then move the render into L209 instead of creating a variable here.